### ISSUE TITLE
Emptying playlist and adding song misbehaves without generating any error

### ISSUE THREAD

**Issue by pmphfm** [suragch/audio_playlist_flutter_demo#2]
Hi,

I used your final example. Ran the demo in chrome. Clicked "-" button to remove all the added songs. When playlist was empty, I added a new song by clicking "+". Tried to play newly added song. But it doesn't play the song. That seems like an unexpected behavior. How do I resolve the issue?

Something similar happens when I use Pixel 4 Api 30 emulator to run the demo.

Thanks,
Pmp

**Comment by suragch** [comment]
Yes, that does sound like unexpected behavior. I've been a pretty busy recently, so I haven't had a chance to work on this, but if you or anyone else finds the solution to the bug, could you leave a comment or make a pull request?
